include/gsl/gsl_byte

```diff
@@ -37,10 +37,30 @@
 #pragma push_macro("noexcept")
 #define noexcept /*noexcept*/
 #endif           // _MSC_VER <= 1800
+
+#if _HAS_STD_BYTE 
+
```

Comment:
Maybe `#if defined(_HAS_STD_BYTE) && _MSC_VER >= 1911` and then educate people that the `_HAS_STD_BYTE` might go away?